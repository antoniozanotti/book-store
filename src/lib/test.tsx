import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
