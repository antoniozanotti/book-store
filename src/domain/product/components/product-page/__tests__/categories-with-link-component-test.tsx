import { render, screen } from "@/lib/test";
import { CategoriesWithLinkComponent } from "../categories-with-link-component";
import { categoriesWithPageMock } from "@/domain/category/__tests__/categories-with-page-mock";

describe("CategoriesWithLinkComponent", () => {
  it("should render", async () => {
    const { container } = render(<CategoriesWithLinkComponent categories={categoriesWithPageMock} />);
    expect(container).toMatchSnapshot();
  });
});