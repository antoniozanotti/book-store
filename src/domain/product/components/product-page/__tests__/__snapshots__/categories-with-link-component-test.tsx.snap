// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesWithLinkComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <a
      class="underline"
      href="/category/arts-and-entertainment"
      title="Arts and Entertainment"
    >
      Arts and Entertainment
    </a>
    , 
    <a
      class="underline"
      href="/category/business-and-technology"
      title="Business and Technology"
    >
      Business and Technology
    </a>
    , 
    <a
      class="underline"
      href="/category/graphic-novels"
      title="Graphic Novels"
    >
      Graphic Novels
    </a>
    , 
    <a
      class="underline"
      href="/category/history-biography-and-social-science"
      title="History, Biography, and Social Science"
    >
      History, Biography, and Social Science
    </a>
    , 
    <a
      class="underline"
      href="/category/home-and-lifestyle"
      title="Home and Lifestyle"
    >
      Home and Lifestyle
    </a>
    , 
    <a
      class="underline"
      href="/category/horror"
      title="Horror"
    >
      Horror
    </a>
    , 
    <a
      class="underline"
      href="/category/languages-and-travel"
      title="Languages and Travel"
    >
      Languages and Travel
    </a>
    , 
    <a
      class="underline"
      href="/category/lgbtqia"
      title="LGBTQIA+"
    >
      LGBTQIA+
    </a>
    , 
    <a
      class="underline"
      href="/category/literature-and-poetry"
      title="Literature and Poetry"
    >
      Literature and Poetry
    </a>
    , 
    <a
      class="underline"
      href="/category/mystery"
      title="Mystery"
    >
      Mystery
    </a>
    , 
    <a
      class="underline"
      href="/category/romance"
      title="Romance"
    >
      Romance
    </a>
    , 
    <a
      class="underline"
      href="/category/science-and-math"
      title="Science and Math"
    >
      Science and Math
    </a>
    , 
    <a
      class="underline"
      href="/category/science-fiction-and-fantasy"
      title="Science Fiction and Fantasy"
    >
      Science Fiction and Fantasy
    </a>
    , 
    <a
      class="underline"
      href="/category/spirituality-and-philosophy"
      title="Spirituality and Philosophy"
    >
      Spirituality and Philosophy
    </a>
    , 
    <a
      class="underline"
      href="/category/wellness-sports-and-outdoors"
      title="Wellness, Sports, and Outdoors"
    >
      Wellness, Sports, and Outdoors
    </a>
    , 
    <a
      class="underline"
      href="/category/young-adult"
      title="Young Adult"
    >
      Young Adult
    </a>
  </div>
</div>
`;
